### PR TITLE
feat: add `FieldAnalyser` to `Source` class

### PR DESCRIPTION
**Pull request by y-lakhdar**
Export `FieldAnalyser` class and use it in the `Source` class.

I also converted some basic objects into Maps to simplify the algorithm

### CODE REVIEW

File: src/fieldAnalyser/fieldAnalyser.ts [hunk 1]
```diff
@@ -49,75 +99,58 @@ export class FieldAnalyser {
...
 
-  private getMostProbableType(field: FieldTypeDict): FieldTypes {
-    const [fieldType] = Object.entries(field).reduce((previous, current) => {
-      if (current[1] > previous[1]) {
-        return current;
-      } else if (current[1] < previous[1]) {
-        return previous;
-      } else {
-        return [current, previous].sort(this.typeCompare)[0];
+  private getMostProbableType(field: FieldTypeMap): FieldTypes {
```
Comment (louis-bompart) [inline src/fieldAnalyser/fieldAnalyser.ts:131]: Here's what I think is a simpler implementation of the method:

```suggestion
  private getMostProbableType(field: FieldTypeMap): FieldTypes {
    const sortedType = Array.from(field.entries()).sort((a, b) => {
      const countDiff = a[1] - b[1];
      return countDiff ? countDiff : this.getFieldTypes(a[0], b[0]);
    })

    return sortedType[0][0];
  }
```

**Review by louis-bompart — COMMENTED** [review]
lgtm 👍 few suggestions

File: src/fieldAnalyser/fieldAnalyser.ts (hunk 1 above)
Comment (y-lakhdar) [reply src/fieldAnalyser/fieldAnalyser.ts:131]: True. A little less performant than the reduce implementation but way more readable.
Nice

Comment (y-lakhdar) [reply src/fieldAnalyser/fieldAnalyser.ts:131]: I compared both implementations with 10000 fields to create.
The `reduce` algorithm takes ~8 seconds vs ~12 seconds (algorithm from your example). 
No big difference in relation to the total amount of time to push documents.

**Review by louis-bompart — APPROVED** [review]

**Pull request merged**